tests/testthat/test-dataset-entity.R

```diff
@@ -350,6 +373,12 @@ with_test_authentication({
             expect_true(is.Numeric(ds$name))
         })
 
+        test_that("PK methods work", {
+            expect_null(pk(ds))
+            expect_silent(pk(ds) <- ds$name)
+            expect_equal(pk(refresh(ds)), ds$name)
```

Comment:
Also test NULL assignment (though the `appendDataset` test will exercise that code). And remove the `refresh` here too.